src/c++/perf_analyzer/genai-perf/tests/test_artifacts.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from argparse import Namespace
+from pathlib import Path
+
+import pytest
+from genai_perf.constants import DEFAULT_ARTIFACT_DIR
+from genai_perf.main import create_artifacts_dirs
+
+
+@pytest.fixture
+def mock_makedirs(mocker):
+    return mocker.patch("os.makedirs")
+
+
+def test_create_artifacts_dirs(mock_makedirs):
+    mock_args = Namespace(artifact_dir=Path(DEFAULT_ARTIFACT_DIR))
+    create_artifacts_dirs(mock_args)
+    assert mock_makedirs.called_with(
```

Comment:
Does this do anything? I don't think called_with is a function in the library. You'd probably want something like: 
```
mock_makedirs.assert_any_call(DEFAULT_ARTIFACT_DIR, exist_ok=True)
mock_makedirs.assert_any_call(Path(DEFAULT_ARTIFACT_DIR) / "plots", exist_ok=True)"
```

Reply:
Yeah my bad. I didn't re-check.

Reply:
I will change it.

Reply:
Thanks! Could you make the test fail first to ensure it fails (e.g. try modifying the args in the calls), then modify it to make it pass? Normally, we would do that by writing and running the test before PR changes, but it's a bit of a different process here since we're adding unit testing for functionality that already exists.

The before (failure) vs after (success) process is still important here though.

Reply:
Ok, Sure (y)

Reply:
Ran the test with wrong path and added the screenshot in the description.

Reply:
Great work!

Reply:
Wait, your screenshot uses assert_any_call but this does not. Did you push your new commits?